### PR TITLE
[Synthetics] Private locations

### PR DESCRIPTION
**Pull request by l0k0ms**
### What does this PR do?

Adds Synthetics private locations documentation

### Motivation
PM request

### Preview link

* https://docs-staging.datadoghq.com/gus/private-location/synthetics/private_locations/

### CODE REVIEW

File: content/en/synthetics/private_locations.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+https://cl.ly/2737639336a2/download/Screen%252520Recording%2525202019-08-02%252520at%25252003.09%252520PM.gif
+
+## Setup
+
+The private location worker is shipped as a Docker container, the image can be found here on Dockerhub.
+
+Your private location worker will periodically (by default, every second) pull your test configurations from Datadog’s servers via HTTPS, execute the test depending on the frequency that was defined during the configuration of the test, and return the associated result to Datadog’s servers.
+The behavior of private locations is completely identical to the one of managed locations.
+
+### Via docker run
```
[the remark this answers is not in the document]
Comment (apigirl) [reply content/en/synthetics/private_locations.md]: reconfigure this section into steps

File: content/en/synthetics/private_locations.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+You first need to create the configuration file associated with your private location on your worker. This file can be found when clicking on Add Private Location on Synthetics / Settings / Private Locations.
+
+https://cl.ly/9be225b3c671
+https://cl.ly/9be225b3c671/download/%255B45641e47632b17672af2c9f7493d72f2%255D_Image%2525202019-08-02%252520at%2525207.21.39%252520PM.png
+
+The configuration file contains secrets for private location authentication, test configuration decryption, and test result encryption.
+Those secrets are not stored by Datadog, so you should make sure to store them in a safe place before leaving the view. You will need these secrets again, would you decide to add more workers, or to install workers on another host.
+
+Launch your worker as a standalone container using a Docker run command and the previously created configuration file.
```
Comment (apigirl) [inline content/en/synthetics/private_locations.md]: this should be a step with the cmds in copy pasteable code

File: content/en/synthetics/private_locations.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+Requirements:
+Docker should be installed on your instance to be able to run the worker.
```
Comment (apigirl) [inline content/en/synthetics/private_locations.md]:
```suggestion
- Install Docker on your instance.
```

File: content/en/synthetics/private_locations.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+The private location worker should be able to access https://api.datadoghq.com/api/ (US instances) or https://api.datadoghq.eu/api/ (EU instances) in order to be able to pull test configurations and push test results.
```
Comment (apigirl) [inline content/en/synthetics/private_locations.md]:
```suggestion
- To pull test configurations and push test results, give the private location worker access to one of the Datadog API endpoints: `https://api.datadoghq.com/api/` (for the US instance) or `https://api.datadoghq.eu/api/` (for the EU instance).
```

Comment (apigirl) [reply content/en/synthetics/private_locations.md]: how do you actually do this?

File: content/en/synthetics/private_locations.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+https://cl.ly/9be225b3c671
+https://cl.ly/9be225b3c671/download/%255B45641e47632b17672af2c9f7493d72f2%255D_Image%2525202019-08-02%252520at%2525207.21.39%252520PM.png
+
+The configuration file contains secrets for private location authentication, test configuration decryption, and test result encryption.
+Those secrets are not stored by Datadog, so you should make sure to store them in a safe place before leaving the view. You will need these secrets again, would you decide to add more workers, or to install workers on another host.
+
+Launch your worker as a standalone container using a Docker run command and the previously created configuration file.
+
+Requirements:
```
Comment (apigirl) [inline content/en/synthetics/private_locations.md]:
```suggestion
#### Docker requirements
```

Comment (apigirl) [reply content/en/synthetics/private_locations.md]: should this section be at the beginning of the docker configuration section?

File: content/en/synthetics/private_locations.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+Docker should be installed on your instance to be able to run the worker.
+The private location worker should be able to access https://api.datadoghq.com/api/ (US instances) or https://api.datadoghq.eu/api/ (EU instances) in order to be able to pull test configurations and push test results.
+
+## Security
+
+The private location workers only pull data from Datadog servers. Datadog does not push any sort of data to the workers.
```
Comment (apigirl) [inline content/en/synthetics/private_locations.md]:
```suggestion
The private location workers only pull data from Datadog servers. Datadog does not push data to the workers.
```

File: content/en/synthetics/private_locations.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+The secret access key, which is the one used to authenticate your private location worker to Datadog servers, uses an in-house protocol based on AWS Signature Version 4 protocol.
```
Comment (apigirl) [inline content/en/synthetics/private_locations.md]:
```suggestion
The secret access key, used to authenticate your private location worker to the Datadog servers, uses an in-house protocol based on AWS Signature Version 4 protocol.
```

File: content/en/synthetics/private_locations.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+The test configurations are encrypted asymmetrically. The private key will be used to decrypt the test configurations pulled by the workers from Datadog servers. The public key will be used to encrypt the test results that are being sent from the workers to Datadog servers.
```
Comment (apigirl) [inline content/en/synthetics/private_locations.md]:
```suggestion
The test configurations are encrypted asymmetrically. The private key is used to decrypt the test configurations pulled by the workers from Datadog servers. The public key is used to encrypt the test results that are sent from the workers to Datadog's servers.
```

File: content/en/synthetics/private_locations.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+## How to scale private locations
+
+To scale-out private locations, you would need to add additional workers.
```
Comment (apigirl) [inline content/en/synthetics/private_locations.md]:
```suggestion
To scale private locations, add or remove workers.
```

Comment (apigirl) [reply content/en/synthetics/private_locations.md]: I feel like this doesn't need it's own section, just a note in the Docker section

File: content/en/synthetics/private_locations.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+## Liveness of the probe
+
+You can know more about the liveness of your private locations by looking at the pills displayed:
```
Comment (apigirl) [inline content/en/synthetics/private_locations.md]:
```suggestion
For more information about the liveness of your private locations, see the pills displayed:
```

File: content/en/synthetics/private_locations.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+on your private locations list, in the Settings section,
+on the form when creating a test, below the Private locations section.
+
```
Comment (apigirl) [inline content/en/synthetics/private_locations.md]: separate these out into a list

File: content/en/synthetics/private_locations.md [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+You’ll get a green pill if the private location polled your endpoint less than 5 seconds before loading the settings or create test pages.
```
Comment (apigirl) [inline content/en/synthetics/private_locations.md]: is this likely to happen often? my lack of docker expertise is showing here, but is someone setting this up likely to not know what this means?

Comment (l0k0ms) [reply content/en/synthetics/private_locations.md]: adding the screenshots helped here

File: content/en/synthetics/private_locations.md [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+
+    ```
+    docker run --name synthetics-private-location-worker-<LOCATION_ID> \
+               --rm -v $(pwd)/synthetics-private-location-worker-<LOCATION_ID>.json:/etc/synthetics-private-location-worker-<LOCATION_ID>.json datadog/synthetics-private-location-worker node /datadog-synthetics-agent/dist/worker.js \
+               --config=/etc/synthetics-private-location-worker-<LOCATION_ID>.json
+    ```
+
+    **Note**: To scale a private location, add or remove workers on your host.
+
+4. To pull test configurations and push test results, the private location worker need access to one of the Datadog API endpoints:
```
Comment (MargotLepizzera) [inline content/en/synthetics/private_locations.md]: needS

File: content/en/synthetics/private_locations.md [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+- link: "synthetics/browser_tests"
+  tag: "Documentation"
+  text: "Configure a Browser Test"
+- link: "synthetics/api_tests"
+  tag: "Documentation"
+  text: "Configure an API Test"
+---
+
+<div class="alert alert-warning">
+This feature is in beta. <a href="https://app.datadoghq.com/privatelocations/2019signup">Sign up for the beta</a> to enable Synthetics private locations for your account.
```
Comment (MargotLepizzera) [inline content/en/synthetics/private_locations.md]: Link to form is different if EU instances - should we specify both?

**Comment by MargotLepizzera** [comment]
Added two comments, other than that LGTM 👍

File: content/en/synthetics/private_locations.md (hunk 14 above)
Comment (l0k0ms) [reply content/en/synthetics/private_locations.md]: well spotted.

File: content/en/synthetics/private_locations.md [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+
+Once you created a private location, configuring a Synthetics API or Browser test from a private location is completely identical to the one of Datadog managed locations.
+
+### Create a new private location
+
+1. Go in *Synthetics* -> *Settings* -> *Private Locations* and create a new private location:
+
+    {{< img src="synthetics/private_locations/create_private_location.png" alt="create a private locations" responsive="true" style="width:70%;">}}
+
+2. Get the configuration file associated with your private location on your worker.
```
Comment (apigirl) [inline content/en/synthetics/private_locations.md]:
```suggestion
2. Fill out the Location Details and click **Save and Generate** to generate the configuration file associated with your private location on your worker.
```

File: content/en/synthetics/private_locations.md [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+
+    {{< img src="synthetics/private_locations/private_location.gif" alt="Private locations" responsive="true" >}}
```
Comment (apigirl) [inline content/en/synthetics/private_locations.md]: worried that this gif shows a secret access key - any way to blur that out?

File: config/_default/menus/menus.en.yaml [hunk 17]
```diff
@@ -1014,27 +1014,33 @@ main:
     identifier: "synthetics_browser_tests"
     weight: 2
```
Comment (apigirl) [inline config/_default/menus/menus.en.yaml]: looks like browser tests got removed from the toc somehow? see preview

File: content/en/synthetics/private_locations.md (hunk 16 above)
Comment (l0k0ms) [reply content/en/synthetics/private_locations.md]: the access key used is now deleted.. so i would say it's fine imho.. ping @MargotLepizzera

File: config/_default/menus/menus.en.yaml (hunk 17 above)
Comment (l0k0ms) [reply config/_default/menus/menus.en.yaml]: fixed, it was an id issue well spotted !

File: content/en/synthetics/private_locations.md (hunk 16 above)
Comment (MargotLepizzera) [reply content/en/synthetics/private_locations.md]: Yes, it's fine because the PL was deleted and is not used.

**Comment by MargotLepizzera** [comment]
LGTM 👍

**Pull request merged**